src/app/contract/contractParser.js

```diff
@@ -1,49 +1,40 @@
 'use strict'
 
-var $ = require('jquery')
 var txHelper = require('../execution/txHelper')
 
-module.exports = (contractName, contract, compiledSource) => {
-  return getDetails(contractName, contract, compiledSource)
+module.exports = (contractName, contract) => {
+  return getDetails(contractName, contract)
 }
 
-var getDetails = function (contractName, contract, source) {
+var getDetails = function (contractName, contract) {
   var detail = {}
   detail.name = contractName
   detail.metadata = contract.metadata
-  if (contract.bytecode) {
-    detail.bytecode = contract.bytecode
+  if (contract.evm.bytecode.object) {
+    detail.bytecode = contract.evm.bytecode.object
   }
 
-  detail.interface = contract.interface
+  detail.abi = contract.abi
 
-  if (contract.bytecode) {
-    detail.bytecode = contract.bytecode
-    detail.web3Deploy = gethDeploy(contractName.toLowerCase(), contract['interface'], contract.bytecode)
+  if (contract.evm.bytecode.object) {
+    detail.bytecode = contract.evm.bytecode
+    detail.web3Deploy = gethDeploy(contractName.toLowerCase(), contract.abi, contract.evm.bytecode.object)
 
-    detail.metadataHash = retrieveMetadataHash(contract.bytecode)
+    detail.metadataHash = retrieveMetadataHash(contract.evm.bytecode.object)
     if (detail.metadataHash) {
-      detail.swarmLocation = 'bzz://' + detail.metadataHash
+      detail.swarmLocation = 'bzzr://' + detail.metadataHash
     }
   }
 
   detail.functionHashes = {}
-  for (var fun in contract.functionHashes) {
-    detail.functionHashes[contract.functionHashes[fun]] = fun
+  for (var fun in contract.evm.methodIdentifiers) {
+    detail.functionHashes[contract.evm.methodIdentifiers[fun]] = fun
   }
 
-  detail.gasEstimates = formatGasEstimates(contract.gasEstimates)
+  detail.gasEstimates = formatGasEstimates(contract.evm.gasEstimates)
 
-  if (contract.runtimeBytecode && contract.runtimeBytecode.length > 0) {
-    detail['Runtime Bytecode'] = contract.runtimeBytecode
-  }
-
-  if (contract.opcodes !== undefined && contract.opcodes !== '') {
-    detail['Opcodes'] = contract.opcodes
-  }
-
-  if (contract.assembly !== null) {
-    detail['Assembly'] = formatAssemblyText(contract.assembly, '', source)
```

Comment:
Actually the old assembly still exists, why not keep it?

Reply:
oh ok i readd it

Reply:
I will readd it